xlators/cluster/afr/src/afr-inode-write.c

```diff
@@ -44,6 +44,9 @@ __afr_inode_write_finalize(call_frame_t *frame, xlator_t *this)
     priv = this->private;
     GF_VALIDATE_OR_GOTO(this->name, local->inode, out);
 
+    if ((local->op == GF_FOP_FSYNC) && (local->num_inodelks > 1))
+        local->transaction.disable_delayed_post_op = _gf_true;
+
```

Comment:
You don't want the lock to be released in this case? Maybe we can make it similar to how writes handle conflicting locks. i.e. get inodelk-count for afr domain. If the number of locks is greater than 1 it will automatically take care of releasing and reacquiring locks.

Reply:
> You don't want the lock to be released in this case?

In the function `afr_changelog_post_op` , if `afr_is_delayed_changelog_post_op_needed` returns false, locks will be released if lock->owner list is empty.

I think we could also use the current writes handling conflict logic but since `inode_ctx->afr_lock` was introduced only for writes, I did not want to modify it and used  `local->transaction.disable_delayed_post_op` which was added for fsync. Should we re-use it for fsync too?

Reply:
Yes.